app/lib/features/pins/pages/pin_details_page.dart

```diff
@@ -0,0 +1,286 @@
+            onTap: () => showEditPintTitleDialog(context, ref, pin),
+            child: Text(L10n.of(context).editTitle),
+          ),
+        );
+
+        //EDIT PIN DESCRIPTION MENU ITEM
+        actions.add(
+          PopupMenuItem<String>(
+            key: PinDetailsPage.editBtnKey,
+            onTap: () => showEditPintDescriptionDialog(context, ref, pin),
+            child: Text(L10n.of(context).editDescription),
+          ),
+        );
+      }
+
+      final canRedact = ref.watch(canRedactProvider(pin));
+      if (canRedact.valueOrNull == true) {
+        final roomId = pin.roomIdStr();
+        //DELETE PIN MENU ITEM
+        actions.add(
+          PopupMenuItem<String>(
+            onTap: () => showRedactDialog(
+              context: context,
+              pin: pin,
+              roomId: roomId,
+            ),
+            child: Text(
+              L10n.of(context).removePin,
+              style: TextStyle(color: Theme.of(context).colorScheme.error),
+            ),
+          ),
+        );
+      } else {
+        //REPORT PIN MENU ITEM
+        actions.add(
+          PopupMenuItem<String>(
+            onTap: () => showReportDialog(context, pin),
+            child: Row(
+              children: <Widget>[
+                Icon(
+                  Atlas.warning_thin,
+                  color: Theme.of(context).colorScheme.error,
+                ),
+                const SizedBox(width: 10),
+                Text(L10n.of(context).reportPin),
+              ],
+            ),
+          ),
+        );
+      }
+    }
+
+    return actions;
+  }
+
+  Widget _buildPinHeaderUI(ActerPin pin) {
+    return Padding(
+      padding: const EdgeInsets.symmetric(horizontal: 16),
+      child: Column(
+        mainAxisAlignment: MainAxisAlignment.start,
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: [
+          Row(
+            children: [
+              pinIconUI(),
+              const SizedBox(width: 12),
+              Expanded(
+                child: Column(
+                  mainAxisSize: MainAxisSize.min,
+                  crossAxisAlignment: CrossAxisAlignment.start,
+                  children: [
+                    pinTitleUI(pin),
+                    pinSpaceNameUI(pin),
+                  ],
+                ),
+              ),
+            ],
+          ),
+          pinDescriptionUI(pin),
+        ],
+      ),
+    );
+  }
+
+  Widget pinIconUI() {
+    return Container(
+      height: 50,
+      width: 50,
+      decoration: BoxDecoration(
+        color: Theme.of(context).colorScheme.secondary,
+        borderRadius: const BorderRadius.all(Radius.circular(100)),
+      ),
+      child: const Icon(Atlas.pin),
+    );
+  }
+
+  Widget pinTitleUI(ActerPin pin) {
+    return SelectionArea(
+      child: GestureDetector(
+        onTap: () {
+          final membership =
+              ref.watch(roomMembershipProvider(pin.roomIdStr())).valueOrNull;
```

Comment:
we are in a function call. see https://acter.global/dev/code-style-best-practices/#when-to-use-refread-and-when-refwatch 
```suggestion
              ref.read(roomMembershipProvider(pin.roomIdStr())).valueOrNull;
```

Reply:
> we are in a function call. see https://acter.global/dev/code-style-best-practices/#when-to-use-refread-and-when-refwatch

Due to copy paste code of provider from build to function code, I missed to change it to `read`. Even knowing this code standards I missed it at couple of places. Will take care of it in next PRs. :)